### PR TITLE
Update for NNlib overhaul

### PR DESCRIPTION
**Pull request by staticfloat**
These changes should be pretty straightforward.

### CODE REVIEW

**Comment by MikeInnes** [comment]
Any idea why CI is not working on the master branch, but is working here?

**Comment by staticfloat** [comment]
There is something wrong with the `Manifest.toml`; it can't find a version or something, hits a bad code path and dies a terrible death.  It's "working" here because I updated the `Manifest.toml`.  If you do a `julia --project=.`, then `]update` I bet it will start working again.

File: test/tracker.jl [hunk 1]
```diff
@@ -265,7 +268,8 @@ end
 
 @test @sprintf("%.2f", sum(param([1,2,3]))) == "6.00"
 
-@inferred NNlib.conv(param(rand(10,10,3,2)),randn(Float64,2,2,3,4))
+# This no longer infers cleanly :(
+#@inferred NNlib.conv(param(rand(10,10,3,2)), randn(Float64,2,2,3,4), DenseConvDims((10,10,3,2),(2,2,3,4)))
```
Comment (MikeInnes) [inline test/tracker.jl]: This is a bit unfortunate, anything we can do about it?

**Comment by MikeInnes** [comment]
Wonderful, that appears to have fixed it indeed.

This PR looks good to me but is failing; does it pass locally?

**Comment by staticfloat** [comment]
The tests now pass locally, I think I mixed something up with my last commit.

**Comment by staticfloat** [comment]
The tests are passing on Travis now too.

**Comment by staticfloat** [comment]
@MikeInnes bump

**Comment by staticfloat** [comment]
@MikeInnes bump

**Comment by MikeInnes** [comment]
Sorry for the delay, thanks for this!

**Pull request merged**